index.js

```diff
@@ -97,6 +97,18 @@ module.exports = {
         }
       },
 
+      fetchRevisions: function(context) {
+        var redisDeployClient = this.readConfig('redisDeployClient');
+        var keyPrefix = this.readConfig('keyPrefix');
+
+        this.log('Listing revisions for key: `' + keyPrefix + '`');
+        return Promise.resolve(redisDeployClient.fetchRevisions(keyPrefix))
+          .then(function(revisions){
+            context.revisions = revisions;
```

Comment:
@ghedamat Great work on this.  One thing though, we'd rather that the context wasn't modified directly. If you simple return an object from this function, then the pipeline will merge it in to the context.  Something like:

```javascript
.then(function(revisions) {
  return { revisions: revisions }
})
```

Happy to merge after than change.